### PR TITLE
add tutorial for colab support

### PR DESCRIPTION
**Pull request by JunweiSUN**

### CODE REVIEW

File: docs/en_US/CommunitySharings/NNI_colab_support.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+# Use NNI on Google Colab
+NNI can easily run on Google's colab platform. However, colab doesn't expose its public IP and ports, so by default you can not access NNI's web ui on colab. To solve this, you need a reverse proxy software like `ngrok` or `frp`. This tutorial will show you how to use ngrok to get access to NNI's web ui on colab.
```
Comment (SparkSnail) [inline docs/en_US/CommunitySharings/NNI_colab_support.md:3]: use ngrok to access NNI's web ui

File: docs/en_US/CommunitySharings/NNI_colab_support.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+```
+! ./ngrok authtoken <your-authtoken>
+```
+
+3. Start an NNI trail example on a port greater than 1024, then start ngrok with the same port.
+
+
+```
+! nnictl create --config nni_repo/nni/examples/trials/mnist-pytorch/config.yml --port 5000 & # if you want to use gpu, make sure gpuNum >= 1 in config.yml
+get_ipython().system_raw('./ngrok http 5000 &')
```
Comment (SparkSnail) [inline docs/en_US/CommunitySharings/NNI_colab_support.md:30]: why not use `./ngrok http 5000 &` directly?

File: docs/en_US/CommunitySharings/NNI_colab_support.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+```
+! curl -s http://localhost:4040/api/tunnels # don't change the port number 4040
+```
+
+You will see an url like http://xxxx.ngrok.io after step 4, open this url and you will find NNI's web ui. Have fun :)
+
+## Access WebUI with frp
+
+frp is another reverse proxy software with similar functions. However, frp doesn't provide free public urls, so you may need an server with public IP as an frp server. See [here](https://github.com/fatedier/frp) to know more about how to deploy frp.
```
Comment (SparkSnail) [inline docs/en_US/CommunitySharings/NNI_colab_support.md:44]: an -> a

File: docs/en_US/CommunitySharings/NNI_colab_support.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+2. Register a ngrok account [here](https://ngrok.com/), then connect to your account using your authtoken.
+
+
+```
+! ./ngrok authtoken <your-authtoken>
+```
+
+3. Start an NNI trail example on a port greater than 1024, then start ngrok with the same port.
```
Comment (SparkSnail) [inline docs/en_US/CommunitySharings/NNI_colab_support.md:25]: greater -> bigger

File: docs/en_US/CommunitySharings/NNI_colab_support.md (hunk 1 above)
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/NNI_colab_support.md]: fixed.

File: docs/en_US/CommunitySharings/NNI_colab_support.md (hunk 2 above)
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/NNI_colab_support.md:30]: If use `! ./ngrok http 5000 &` directly, the cell running this command will stuck. Not sure why this happened. Use get_ipython() will solve this problem.

File: docs/en_US/CommunitySharings/NNI_colab_support.md (hunk 3 above)
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/NNI_colab_support.md]: fixed.

File: docs/en_US/CommunitySharings/NNI_colab_support.md (hunk 4 above)
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/NNI_colab_support.md]: fixed.

File: docs/en_US/nnicli.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# NNI Client
+
+NNI client is a python API of `nnictl`, which implements the most common used commands. User can use this API to control their experiments in python code instead of command line. Here is an example:
+
+```
```
Comment (ultmaster) [inline docs/en_US/nnicli.md:5]: Might submit this file in a separate PR.

File: docs/en_US/CommunitySharings/NNI_colab_support.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+
+# Use NNI on Google Colab
+NNI can easily run on Google's colab platform. However, colab doesn't expose its public IP and ports, so by default you can not access NNI's web ui on colab. To solve this, you need a reverse proxy software like `ngrok` or `frp`. This tutorial will show you how to use ngrok to access NNI's web ui on colab.
```
[the remark this answers is not in the document]
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/NNI_colab_support.md:3]: fixed.

File: docs/en_US/CommunitySharings/NNI_colab_support.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+2. Register a ngrok account [here](https://ngrok.com/), then connect to your account using your authtoken.
+
+
+```
+! ./ngrok authtoken <your-authtoken>
+```
+
+3. Start an NNI trail example on a port bigger than 1024, then start ngrok with the same port.
```
[the remark this answers is not in the document]
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/NNI_colab_support.md:25]: fixed.

File: docs/en_US/CommunitySharings/NNI_colab_support.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+
+```
+! nnictl create --config nni_repo/nni/examples/trials/mnist-pytorch/config.yml --port 5000 & # if you want to use gpu, make sure gpuNum >= 1 in config.yml
```
[the remark this answers is not in the document]
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/NNI_colab_support.md:29]: fixed.

File: docs/en_US/CommunitySharings/NNI_colab_support.md (hunk 2 above)
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/NNI_colab_support.md:30]: fixed.

File: docs/en_US/nnicli.md (hunk 5 above)
Comment (JunweiSUN) [reply docs/en_US/nnicli.md:5]: Sure. These files were introduced by mistake.

File: docs/en_US/CommunitySharings/NNI_colab_support.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
+
+# Use NNI on Google Colab
+NNI can easily run on Google Colab platform. However, colab doesn't expose its public IP and ports, so by default you can not access NNI's Web UI on colab. To solve this, you need a reverse proxy software like `ngrok` or `frp`. This tutorial will show you how to use ngrok to access NNI's Web UI on colab.
```
Comment (ultmaster) [inline docs/en_US/CommunitySharings/NNI_colab_support.md:3]: There is still a lot of "colab". Please change them to "Colab".

File: docs/en_US/CommunitySharings/NNI_colab_support.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+4. Check the public url.
+
+
+```
+! curl -s http://localhost:4040/api/tunnels # don't change the port number 4040
+```
+
+You will see an url like http://xxxx.ngrok.io after step 4, open this url and you will find NNI's web ui. Have fun :)
```
Comment (ultmaster) [inline docs/en_US/CommunitySharings/NNI_colab_support.md:40]: Web UI

File: docs/en_US/CommunitySharings/NNI_colab_support.md (hunk 9 above)
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/NNI_colab_support.md]: My bad. Fixed.

File: docs/en_US/CommunitySharings/NNI_colab_support.md (hunk 10 above)
Comment (JunweiSUN) [reply docs/en_US/CommunitySharings/NNI_colab_support.md]: fixed.
